src/components/ExportButton/ExportButton.js

```diff
@@ -0,0 +1,194 @@
+    minWidth: 85,
+  },
+  button: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    padding: '5px',
+    height: 35,
+    width: 85,
+    '&:hover': {
+      color: '#eb0000',
+    },
+  },
+  loading: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+}));
+
+function ExportButton({
+  exportFormat,
+  exportScale,
+  exportSize,
+  exportCoordinates,
+  exportZoom,
+  exportExtent,
+  children,
+}) {
+  const classes = useStyles();
+  const map = useSelector((state) => state.app.map);
+  const topic = useSelector((state) => state.app.activeTopic);
+  const layerService = useSelector((state) => state.app.layerService);
+  const { t } = useTranslation();
+  const [isLoading, setLoading] = useState(false);
+
+  return (
+    <div className={classes.buttonWrapper}>
+      <CanvasSaveButton
+        className={classes.button}
+        title={t('Als PNG speichern')}
+        style={{
+          pointerEvents: isLoading ? 'none' : 'default',
+          opacity: isLoading ? 0.3 : 1,
+          width: 'auto',
+        }}
+        extraData={generateExtraData(layerService, true)}
+        autoDownload={false}
+        format="image/jpeg"
+        onSaveStart={() => {
+          setLoading(true);
+          return getMapHd(
+            map,
+            layerService,
+            exportScale,
+            exportCoordinates,
+            exportSize,
+            exportZoom,
+            exportExtent,
+          );
+        }}
+        onSaveEnd={async (mapToExport, canvas) => {
+          clean(mapToExport, map, layerService);
+
+          // add the image to a newly created PDF
+          const doc = new JsPDF({
+            orientation: 'landscape',
+            unit: 'pt',
+            format: exportFormat,
+          });
+
+          // Add map image
+          const ctx = canvas.getContext('2d');
+          ctx.scale(1 / exportScale, 1 / exportScale);
+          doc.addImage(canvas, 'JPEG', 0, 0, exportSize[0], exportSize[1]);
+
+          /**
+           * CAUTION: SVG parsing and dynamic value insertion will break if the legend SVG tree and tag IDs
+           * are not maintained. If changes in the legend SVG are necessary, make sure the tree and IDs are maintained
+           * It is also recommended to use inkscape (Adobe illustrator SVG won't work out-of-the-box
+           * without major alterations)
+           */
+          // Fetch local svg
+          const svgString = await fetch(legend).then((response) =>
+            response.text(),
+          );
+
+          // svgson parse string to json to access values
+          const svgJson = await parseSvg(svgString).then((json) => json);
+          const legendElements = svgJson.children.find(
+            (tag) => tag.attributes.id === 'legend_elements',
+          ).children;
+
+          // Set date DE
+          legendElements.find(
+            (element) => element.attributes.id === 'date_DE',
```

Comment:
This looks very brittle. How about adding some unique placeholder strings in the SVG file which can be automatically replaced by values configured in `exportConfig` using some regex?

Reply:
I ditched the svgson stuff, which was really unnecessary considering your approach of replacing unique strings.